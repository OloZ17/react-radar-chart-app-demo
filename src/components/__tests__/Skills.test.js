import React from 'react';
import { render } from '@testing-library/react';
import { Skills } from '../index';

describe('Testing Skills() Component', () => {
  const skills = [
    'Thing 1',
    'Thing 2',
    'Thing 3',
    'Thing 4',
    'Thing 5',
    'Thing 6',
  ];

  const handleChangeSkills = jest.fn();
  const addSkill = jest.fn();
  const removeSkill = jest.fn();

  it('should take a snapshot', () => {
    const { asFragment } = render(
      <Skills
        skills={skills}
        handleChangeSkills={handleChangeSkills}
        addSkill={addSkill}
        removeSkill={removeSkill}
      />
    );
    expect(asFragment(<Skills />)).toMatchSnapshot();
  });
});
