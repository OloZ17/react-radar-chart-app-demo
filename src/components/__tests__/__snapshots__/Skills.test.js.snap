// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Skills() Component should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <h3>
      Skills
    </h3>
    <div
      class="form-inline form-group mb-2"
    >
      <label>
        Skill 1
      </label>
      <input
        name="skill"
        style="padding: 2px; text-align: left; margin-right: 5px; margin-left: 5px;"
        type="text"
        value="Thing 1"
      />
      <button
        class="btn btn-danger"
        type="button"
      >
        Remove
      </button>
    </div>
    <div
      class="form-inline form-group mb-2"
    >
      <label>
        Skill 2
      </label>
      <input
        name="skill"
        style="padding: 2px; text-align: left; margin-right: 5px; margin-left: 5px;"
        type="text"
        value="Thing 2"
      />
      <button
        class="btn btn-danger"
        type="button"
      >
        Remove
      </button>
    </div>
    <div
      class="form-inline form-group mb-2"
    >
      <label>
        Skill 3
      </label>
      <input
        name="skill"
        style="padding: 2px; text-align: left; margin-right: 5px; margin-left: 5px;"
        type="text"
        value="Thing 3"
      />
      <button
        class="btn btn-danger"
        type="button"
      >
        Remove
      </button>
    </div>
    <div
      class="form-inline form-group mb-2"
    >
      <label>
        Skill 4
      </label>
      <input
        name="skill"
        style="padding: 2px; text-align: left; margin-right: 5px; margin-left: 5px;"
        type="text"
        value="Thing 4"
      />
      <button
        class="btn btn-danger"
        type="button"
      >
        Remove
      </button>
    </div>
    <div
      class="form-inline form-group mb-2"
    >
      <label>
        Skill 5
      </label>
      <input
        name="skill"
        style="padding: 2px; text-align: left; margin-right: 5px; margin-left: 5px;"
        type="text"
        value="Thing 5"
      />
      <button
        class="btn btn-danger"
        type="button"
      >
        Remove
      </button>
    </div>
    <div
      class="form-inline form-group mb-2"
    >
      <label>
        Skill 6
      </label>
      <input
        name="skill"
        style="padding: 2px; text-align: left; margin-right: 5px; margin-left: 5px;"
        type="text"
        value="Thing 6"
      />
      <button
        class="btn btn-danger"
        type="button"
      >
        Remove
      </button>
    </div>
    <div
      class="button-section"
    >
      <button
        class="btn btn-primary"
        type="button"
      >
        Add
      </button>
    </div>
  </div>
</DocumentFragment>
`;
